tasks/managed/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -130,72 +136,153 @@ spec:
+            | jq -r '.sign_index_image' |tr -d "\n")
+          mustPublishIndexImage=$(jq -r '.genericResult | fromjson' <<< "${results}" \
+            | jq -r '.publish_index_image' |tr -d "\n")
+          fbc_opt_in=$(jq -r '.genericResult | fromjson' <<< "${results}" | jq -cr '.fbc_opt_in')
 
-        jq -r '.indexImageDigests' <<< "${results}" |  tee "$(results.indexImageDigests.path)"
+          # Store the results in Tekton's results files
+          echo -en "${mustPublishIndexImage}" | tee "$(results.mustPublishIndexImage.path)"
+          echo -en "${mustSignIndexImage}" | tee "$(results.mustSignIndexImage.path)"
+          echo -en "${fbc_opt_in}" | tee "$(results.isFbcOptIn.path)"
 
-        jq -r '.iibLog' <<< "${results}"
-        RC="$(jq -r '.exitCode' <<< "${results}")"
+          # this results below should be depracated
+          conditions=$(kubectl get internalrequest "${internalRequest}" \
+            -o jsonpath='{.status.conditions[?(@.type=="Succeeded")]}')
+          jq '.reason // "Unset"'  <<< "${conditions}" | tee "$(results.requestReason.path)"
+          jq '.message // "Unset"' <<< "${conditions}" | tee "$(results.requestMessage.path)"
+          jq -r '.indexImageDigests' <<< "${results}" |  tee "$(results.indexImageDigests.path)"
 
-        # Summarize what happened for the human user.
-        if [ "$mustPublishIndexImage" = "true" ]; then
-          echo "Index image will be published."
-        elif [ "$fbc_opt_in" = "false" ]; then
-          echo "Index image will not be published because fbc_opt_in is set to false in Pyxis."
-          echo "If this is the first time you are releasing, make sure you request fbc_opt_in in Pyxis."
-        elif [ "${staged_index}" = "true" ]; then
-          echo "Index image will not be published because this is a staging release."
+          # the following ones are used to help debugging in case of failure
+          jq -r '.iibLog' <<< "${results}"
+          RC="$(jq -r '.exitCode' <<< "${results}")"
+
+          # Summarize what happened for the human user. Although the continuation of the pipeline will depend on all
+          # components having the same flag values, we should display them to help the user to understand which of the
+          # component might be misconfigured.
+          if [ "$mustPublishIndexImage" = "true" ]; then
+            echo "Index image will be published."
+          elif [ "$fbc_opt_in" = "false" ]; then
+            echo "Index image will not be published because fbc_opt_in is set to false in Pyxis."
+            echo "If this is the first time you are releasing, make sure you request fbc_opt_in in Pyxis."
+          elif [ "${staged_index}" = "true" ]; then
+            echo "Index image will not be published because this is a staging release."
+          else
+            echo "Index image will not be published for an unspecified reason."
+          fi
+
+          # if it any of the IR failed, it should stop.
```

Comment:
I found this a bit confusing. I would either remove this or say "if the IR failed, we should stop".

Because the current comment sounds like what we do elsewhere where we just make a not of a non-zero exit and at the end we fail if at least one failed. That's not what's happening here - here we check after each IR and stop if it failed.

Also, that's what you intended, right? To fail right away if an IR failed. Have you considered doing all of the IRs anyway and fail only after the last one? There's probably no reason for that here, right?

Reply:
Yeah. I think I can remove. 
The behavior is - if the running IR fails, it should stop the pipeline - I don' t think for the moment we want to give the option to partially publish the catalog in case one of the builds in IIB not succeeded.
Although in the near future we want the IRs running in parallel, as IIB can manage this.
The reason to not do like this is on pressure of time. My impression is that would add some extra complexity to it now, and delay the delivery, and this can be improved in the subsequent sprints.

Reply:
The task script is currently malformatted: https://github.com/konflux-ci/release-service-catalog/pull/930/files#r2111677861

So I will check this later.